### PR TITLE
V5.3.2 audit mergeback

### PR DESCRIPTION
**Pull request by charlesjhongc**
This is a mergeback PR that includes `v5.3.2` fixes during first internal audit.

### CODE REVIEW

File: doc/strategies/RFQv2.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+| Field               |  Type   | Description                                                                        |
+| ------------------- | :-----: | ---------------------------------------------------------------------------------- |
+| taker               | address | The address of the taker of an offer. An offer can be filled only by this address. |
+| maker               | address | The address of the maker of an offer.                                              |
+| takerToken          | address | The address of taker token.                                                        |
+| takerTokenAmount    | uint256 | The amount of taker token.                                                         |
+| makerToken          | address | The address of maker token.                                                        |
+| makerTokenAmount    | uint256 | The amount of maker token.                                                         |
+| minMakerTokenAmount | uint256 | The minimum amount of maker token. Should be identical to makerTokenAmount here.   |
```
Comment (NIC619) [inline doc/strategies/RFQv2.md:26]: This is removed?

File: doc/strategies/RFQv2.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+| expiry              | uint256 | The timestamp of the expiry.                                                       |
+| salt                | uint256 | A random number included in an offer to avoid replay attack.                       |
+
+## Signature and Relayer
```
Comment (NIC619) [inline doc/strategies/RFQv2.md:30]: Relayer is not mentioned in this section?

File: doc/strategies/RFQv2.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+The maker of an offer should provide signature of the `Offer` struct to authorize the maker side of the trade. While the taker should sign the `RFQOrder` struct which is the `Offer` struct plus `recipient` and `feeFactor`.
+
+## Fee & Relayer
+
+Some portion of maker asset of an offer will be deducted as protocol fee. The fee will be transfered to `feeCollector` during the settlement. Each offer may have different fee factor which is provided by Tokenlon's quoting system. Only those trades with proper `feeFactor` will be accepted by Tokenlon's relayers. An user could submit the trade without relayer and therefore the `feeFactor` is zero in such case.
```
Comment (NIC619) [inline doc/strategies/RFQv2.md:36]:
```suggestion
Some portion of maker asset of an offer will be deducted as protocol fee. The fee will be transfered to `feeCollector` during the settlement. Each offer may have different fee factor which is provided by Tokenlon's quoting system. Only those trades with proper `feeFactor` will be accepted by Tokenlon's relayers. A user could submit the trade without relayer and therefore the `feeFactor` is zero in such case.
```

File: doc/strategies/RFQv2.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+## WETH
+
+If the `makerToken` of an order is WETH, then RFQv2 contract will unwrap it and send ETH to the `recipient`.
```
Comment (NIC619) [inline doc/strategies/RFQv2.md:40]: It will do the same for `takerToken`?

**Comment by charlesjhongc** [comment]
@NIC619 I forgot to update the doc of RFQv2. Update it in #191 and will merge into `v5.3.2-audit` first so this one will be updated as well.

**Review by NIC619 — DISMISSED** [review]

**Review by pilagod — APPROVED** [review]

File: doc/strategies/RFQv2.md (hunk 4 above)
Comment (charlesjhongc) [reply doc/strategies/RFQv2.md]: Added in #191

File: doc/strategies/RFQv2.md (hunk 3 above)
Comment (charlesjhongc) [reply doc/strategies/RFQv2.md]: #191

File: doc/strategies/RFQv2.md (hunk 2 above)
Comment (charlesjhongc) [reply doc/strategies/RFQv2.md]: #191

File: doc/strategies/RFQv2.md (hunk 1 above)
Comment (charlesjhongc) [reply doc/strategies/RFQv2.md]: #191

**Review by NIC619 — APPROVED** [review]

**Review by pilagod — APPROVED** [review]

**Pull request merged**